m4/curl-rustls.m4

```diff
@@ -141,6 +141,7 @@ if test "x$OPT_RUSTLS" != xno; then
       ssl_msg="rustls"
       AC_DEFINE(USE_RUSTLS, 1, [if rustls is enabled])
       AC_SUBST(USE_RUSTLS, [1])
+      USE_RUSTLS="yes"
```

Comment:
Looking closer this was already being set on L140, so I'm less sure I understand why this fixes the issue.

Reply:
maybe `AC_DEFINE(USE_RUSTLS` being after... ?

Reply:
but if I apply only this (patch), it does fix it for me as well, and if I don't apply this at all, it breaks it - so, confirmed working:)

Reply:
Before the pkg-config changes [the success case configuration](https://github.com/curl/curl/blob/b564a5f5d5280387a88025c5f90260017847add4/m4/curl-rustls.m4#L55-L58) looked like:

1. AC_DEFINE...
2. AC_SUBST...
3. RUSTLS_ENABLED=1
4. USE_RUSTLS=yes

I think that order of operations was preserved in https://github.com/curl/curl/commit/647e86a3efe1eea7a2a456c009cfe1eb55fe48eb, but in 9c4209837094781d5eef69ae6bcad0e86b64bf99 the order changed to:

 1. USE_RUSTLS="yes"
 2.  AC_DEFINE...
 3.  AC_SUBST...
 4.  RUSTLS_ENABLED=1

Reply:
I updated my patch to _move_ the existing `USE_RUSTLS="yes"` instead of adding a second. The commit message & PR title are also updated to reflect that it's the order of operations here that seems to matter.

Reply:
confirmed working for me with the repositioning too

Reply:
Thank you for testing